### PR TITLE
[JENKINS-29195] Add basic support and functionality for HP-UX

### PR DESCRIPTION
**Pull request by pjanouse**
Please review and merge. Thx Pavel

### CODE REVIEW

File: src/main/java/org/jenkinsci/plugins/proccleaner/ProcCleaner.java [hunk 1]
```diff
@@ -60,6 +60,19 @@ public static int getpid(){
     }
 
     /**
+     *
+     * @return OS name with lowered cases
+     */
+    public static String getOsName() {
+        String os = System.getProperty("os.name", "");
+
+        if (os.length() > 0)
```
Comment (olivergondza) [inline src/main/java/org/jenkinsci/plugins/proccleaner/ProcCleaner.java]: Why is empty string any special? Do you mean `os == null`?

Comment (pjanouse) [inline src/main/java/org/jenkinsci/plugins/proccleaner/ProcCleaner.java]: Oliver, your are right, the condition isn't necessary in current impl., fixing...

Pavel

File: src/main/java/org/jenkinsci/plugins/proccleaner/PsBasedHPUXProcessTree.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        pb.redirectErrorStream(true);
+        Process proc = pb.start();
+        reader = new BufferedReader(new InputStreamReader(proc.getInputStream()));
+        proc.waitFor();
+        PsBasedProcessTree ptree = new PsBasedHPUXProcessTree();
+        String line = reader.readLine(); // first line should be "UID PID PPID C STIME TTY TIME COMMAND" - skip it
+        if (!line.matches("^\\s*UID\\s*PID\\s*PPID\\s*C\\s*STIME\\s*TTY\\s*TIME\\s*COMMAND\\s*$")) {
+            return null;
+        }
+        boolean unsupported_ps = false;
```
Comment (olivergondza) [inline src/main/java/org/jenkinsci/plugins/proccleaner/PsBasedHPUXProcessTree.java]: This is unused.

**Comment by olivergondza** [comment]
:+1:

File: src/main/java/org/jenkinsci/plugins/proccleaner/PsProcessHPUX.java [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+        super(pid, ppid, args, ptree);
+    }
+
+    @Override
+    public void kill(int signum) {
+        // jna doesn't support HP-UX yet
+        //LIBC.kill(super.getPid(), signum);
+        try {
+            String[] cmd = {"kill", "-" + signum, ((Integer) getPid()).toString()};
+            BufferedReader reader;
```
Comment (olivergondza) [inline src/main/java/org/jenkinsci/plugins/proccleaner/PsProcessHPUX.java]: unused

**Comment by vjuranek** [comment]
Merged, thanks!

Contained in `proc-cleaner 1.1` relased right now.

**Comment by olivergondza** [comment]
I should have mentioned Pavel is working on a test case. Thanks for merging.

**Pull request closed without merging**